main.go

```diff
@@ -204,55 +224,62 @@ func setupRayClusterController(mgr ctrl.Manager, cfg *config.CodeFlareOperatorCo
 	return rayClusterController.SetupWithManager(mgr)
 }
 
-// +kubebuilder:rbac:groups="apiextensions.k8s.io",resources=customresourcedefinitions,verbs=get;list;watch
-
 func waitForRayClusterAPIandSetupController(ctx context.Context, mgr ctrl.Manager, cfg *config.CodeFlareOperatorConfiguration, isOpenShift bool, certsReady chan struct{}) {
-	crdClient, err := apiextensionsclientset.NewForConfig(mgr.GetConfig())
-	exitOnError(err, "unable to create CRD client")
-
-	crdList, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
-	exitOnError(err, "unable to list CRDs")
-
-	if slices.ContainsFunc(crdList.Items, func(crd apiextensionsv1.CustomResourceDefinition) bool {
-		return crd.Name == "rayclusters.ray.io"
-	}) {
+	if isAPIAvailable(ctx, mgr, rayclusterAPI) {
 		exitOnError(setupRayClusterController(mgr, cfg, isOpenShift, certsReady), "unable to setup RayCluster controller")
+	} else {
+		waitForAPI(ctx, mgr, rayclusterAPI, func() {
+			exitOnError(setupRayClusterController(mgr, cfg, isOpenShift, certsReady), "unable to setup RayCluster controller")
+		})
 	}
+}
 
-	retryWatcher, err := retrywatch.NewRetryWatcher(crdList.ResourceVersion, &cache.ListWatch{
-		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-			return crdClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
-		},
-		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-			return crdClient.ApiextensionsV1().CustomResourceDefinitions().Watch(ctx, metav1.ListOptions{})
-		},
-	})
-	exitOnError(err, "unable to create retry watcher")
-
-	defer retryWatcher.Stop()
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case event := <-retryWatcher.ResultChan():
-			switch event.Type {
-			case watch.Error:
-				exitOnError(apierrors.FromObject(event.Object), "error watching for RayCluster API")
+func setupAppWrapperComponents(ctx context.Context, cancel context.CancelFunc, mgr ctrl.Manager,
+	cfg *config.CodeFlareOperatorConfiguration, certsReady chan struct{}) error {
+	if cfg.AppWrapper == nil || !ptr.Deref(cfg.AppWrapper.Enabled, false) {
+		setupLog.Info("Embedded AppWrapper controller is disabled by config")
+		go func() {
+			<-certsReady
+			setupLog.Info("Setting up mock AppWrapper webhooks")
+			exitOnError(controllers.SetupMockAppWrapperWebhooks(mgr), "unable to setup AppWrapper webhooks")
```

Comment:
Based on our discussion, I'm thinking it may be simpler and better from a UX standpoint to just not setup that NOOP webhook. Creation of AppWrappers will fail-fast according to the failure policy set in the the webhook configuration, so it'll be clear for clients / users that it's not going to work as opposed to let the resources be created but not reconciled. WDYT? That assumes the external controller shares the same webhook configuration. If that's not the case, then the NOOP webhook would indeed be needed.

Reply:
Let's try the simplest approach for now so we can get this merge.  I'll take out the noop hook.

We'll verify that we can modify the webhook of a deployed codeflare operator to connect to the external controller's endpoints next week and if that doesn't work, we can figure out what to do then.

Reply:
Done